### PR TITLE
[Doc] Add troubleshooting and best practice sections

### PR DESCRIPTION
**Pull request by karenzone**
Add troubleshooting and FAQ sections to collect tips and other tactical info.
Issue #9351

### CODE REVIEW

**Comment by karenzone** [comment]
FOR DISCUSSION:

@jsvd  Here's a rough draft of what a troubleshooting/best practices section for LS might look like. Currently, most of the content is based on Kafka issues that @original-brownbear wrote up because was answering the same questions again and again. 

I'm thinking that we can help users--and help ourselves--if we have a central place to document tips and "gotchas."  We can add targeted troubleshooting info to this section rather than having it sprinkled through the doc where it's likely to be overlooked.  

@dede I'd appreciate your input as well.  I don't want to dump random issues in without any structure. However, I'm concerned that the structure I've started might be problematic from a web redirect standpoint as content grows and structure (headings, categories, etc) morphs to accommodate new info.

File: docs/static/faq.asciidoc [hunk 1]
```diff
@@ -0,0 +1,64 @@
+[[faq]] 
+== Frequently Asked Questions (FAQ)
+
+We will be adding more questions and answers, so please check back soon.
```
Comment (jsvd) [inline docs/static/faq.asciidoc]: should we link to this file on github and ask folks to open prs about questions they think we should have here?

File: docs/static/faq.asciidoc [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+[float]
+[[faq-kafka-offset-commit]]
+===== Does Kafa Input commit offsets only for events that have passed the pipeline fully?
+No, we can’t make the guarantee. Offsets are committed to Kafka periodically. If
+writes to the PQ are slow/blocked, offsets for events that haven’t safely
+reached the PQ can be committed. 
+
+
+
```
Comment (jsvd) [inline docs/static/faq.asciidoc]: let's add some way for users to provide feedback, maybe a link to logstash github issues?

File: docs/static/troubleshooting.asciidoc [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+
+[float] 
+[[ts-temp-dir]] 
+=== Inaccessible temp directory
+
+Certain versions of the JRuby runtime and libraries
+in certain plugins (the Netty network library in the TCP input, for example) copy
+executable files to the temp directory. This situation causes subsequent failures when
+`/tmp` is mounted `noexec`. 
+
```
Comment (jsvd) [inline docs/static/troubleshooting.asciidoc]: we should fetch a sample error here too

File: docs/static/troubleshooting.asciidoc [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+[float] 
+[[ts-429]] 
+=== Error response code 429
+
+A `429` message indicates that an application is busy handling other requests. For
+example, Elasticsearch throws a `429` code to notify Logstash (or other indexers)
+that the bulk failed because the ingest queue is full. Any documents that
+weren't processed should be retried.
+
+TBD:  Does Logstash retry? Should the user take any action?
```
Comment (jsvd) [inline docs/static/troubleshooting.asciidoc]: logstash will retry forever here. potential actions would be to check how es is behaving, we can link to the elasticsearch monitoring docs https://www.elastic.co/guide/en/elasticsearch/reference/current/es-monitoring.html
and also maybe some non-xpack apis too: https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-stats.html

File: docs/static/troubleshooting.asciidoc [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+Instead of:
+
+-----
+bin/logstash -e 'input { stdin { } } output { stdout {} }'
+-----
+
+Use this format on Windows systems:
+
+-----
+bin/logstash -e "input { stdin { } } output { stdout {} }"
```
Comment (jsvd) [inline docs/static/troubleshooting.asciidoc]: also you need to invert that forward slash: `bin\logstash` as paths in windows are separated by backslash

File: docs/static/faq.asciidoc (hunk 1 above)
Comment (karenzone) [reply docs/static/faq.asciidoc]: I like that idea.  After I publish this topic, I'll rework the PR links directly to the source file. For now, I linked it to the repo as a placeholder.

**Pull request closed without merging**